legal-api/src/legal_api/services/filings/validations/agm_extension.py

```diff
@@ -0,0 +1,181 @@
+                        'path': f'{AGM_EXTENSION_PATH}/prevAgmRefDate'})
+        else:
+            total_approved_ext = get_int(filing, f'{AGM_EXTENSION_PATH}/totalApprovedExt')
+            extension_duration = get_int(filing, f'{AGM_EXTENSION_PATH}/extensionDuration')
+            if total_approved_ext != 6 or extension_duration != 6:
+                msg.append({'error': babel('Fail to grant extension.')})
+    else:
+        # subsequent AGM, second extension or more
+        if not (curr_ext_expire_date_str := get_str(filing, f'{AGM_EXTENSION_PATH}/expireDateCurrExt')):
+            return [{'error': 'Expiry date for current extension is required.',
+                     'path': f'{AGM_EXTENSION_PATH}/expireDateCurrExt'}]
+
+        curr_ext_expire_date =\
+            LegislationDatetime.as_legislation_timezone_from_date_str(curr_ext_expire_date_str).date()
+
+        allowable_ext_date = prev_agm_ref_date + relativedelta(months=12)
+        now = LegislationDatetime.datenow()
+
+        if curr_ext_expire_date >= allowable_ext_date:
+            msg.append({'error': 'Company has received the maximum 12 months of allowable extensions.',
+                        'path': f'{AGM_EXTENSION_PATH}/expireDateCurrExt'})
+        elif now > curr_ext_expire_date + relativedelta(days=5):
+            msg.append({'error': 'Allotted period to request extension has expired.',
+                        'path': f'{AGM_EXTENSION_PATH}/expireDateCurrExt'})
+        else:
+            total_approved_ext = get_int(filing, f'{AGM_EXTENSION_PATH}/totalApprovedExt')
+            extension_duration = get_int(filing, f'{AGM_EXTENSION_PATH}/extensionDuration')
+
+            expected_total_approved_ext, expected_extension_duration =\
+                _calculate_granted_ext(curr_ext_expire_date, prev_agm_ref_date)
+
+            if expected_total_approved_ext != total_approved_ext or\
+                    expected_extension_duration != extension_duration:
+                msg.append({'error': babel('Fail to grant extension.')})
+
+    return msg
+
+
+def intended_agm_date_validation(filing: Dict) -> list:
+    """Validate intended AGM date."""
+    msg = []
+    intended_agm_date_str = get_str(filing, f'{AGM_EXTENSION_PATH}/intendedAgmDate')
+    curr_ext_expire_date_str = get_str(filing, f'{AGM_EXTENSION_PATH}/expireDateCurrExt')
+    if intended_agm_date_str and curr_ext_expire_date_str:
+        intended_agm_date =\
+            LegislationDatetime.as_legislation_timezone_from_date_str(intended_agm_date_str).date()
+        curr_ext_expire_date =\
+            LegislationDatetime.as_legislation_timezone_from_date_str(curr_ext_expire_date_str).date()
+
+        if intended_agm_date > curr_ext_expire_date:
+            msg.append({'error': 'Intended AGM date should not be greater than current extension expiry date.',
+                        'path': f'{AGM_EXTENSION_PATH}/intendedAgmDate'})
```

Comment:
so when i said `intended_agm_date` should not be greater than the extension expiry date, I probably wasn't clear enough.  I meant the `intended_agm_date` should not be greater than whatever new expiry date you would be granting for the extension being requested.

Reply:
intended_agm_date validation can be removed as per Mihai's comment in the ticket.